src/EditorFeatures/Core.Wpf/SymbolSearch/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -232,27 +232,31 @@ private async Task<TimeSpan> DownloadFullDatabaseAsync()
                     var (succeeded, delay) = await DownloadFullDatabaseWorkerAsync().ConfigureAwait(false);
                     if (succeeded)
                     {
-                        await _service._progressService.OnDownloadFullDatabaseSucceededAsync(_service._updateCancellationToken).ConfigureAwait(false);
+                        // Make sure the completion notification is sent even if cancellation is requested
+                        await _service._progressService.OnDownloadFullDatabaseSucceededAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
i'm not certain i understnd how this helps.  say we cancel on the line right above this.  i.e. in DownloadFullDatabaseWorkerAsync.  Won't we also not update any particular notification that is already being displayed?

Reply:
That would result in a cancellation exception, and report the cancellation.

Reply:
I am not sure which code run where. but from index, looks like call can be from OOP and task center is in VS, then if cancellation token happen before it is sent to wire, the call itself will be dropped and cancellation exception will be thrown. and I am not sure how this code is written to handle that case.

by looking at the VS side code, it looks like cancellation can never be thrown if it ran in proc, which is not true anymore in OOP case since framework can throw cancellation.

I would either just remove cancellation parameter from Succeeded and Failed or make code to handle cancellation properly thrown from OOP.

Reply:
looking this.
http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs,18

no one uses cancellation token?

Reply:
my guess what is happening is. this code runs on OOP and cancellation is true, so StreamJsonRpc throws cancellation exception and never call the other side and the other side gets RemoteInvocationException (or CancellationException based on whether the other side used JsonRpcEx or not to workaround StreamJsonRpc's cancellation exception issue) but that one goes to void and ignored.

http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Wpf/SymbolSearch/SymbolSearchUpdateEngineFactory.cs,94

doesn't have any exception handling..

Reply:
> so StreamJsonRpc throws cancellation exception and never call the other side and the other side gets RemoteInvocationException (or CancellationException based on whether the other side used JsonRpcEx or not to workaround StreamJsonRpc's cancellation exception issue) but that one goes to void and ignored.

When does this happen?  When does StreamJsonRpc throw cancellation exception?  Are there docs we can look at to understand the contract here?  This code was originally written to be in-proc using VS/Roslyn threading/cancellation semantics.  If those semantics are different here we'd def have to fix things up appropriately.

--

Note: what's super weird and hard to understand is that nearly all the codepaths i trace back now lead to a CancellationToken.None.  So we appear to have just broken cancellation here pretty significantly.  This is also strange because it means i also don't understand how @sharwell is seeing what he's seeing.